application/controllers/template_controller.py

```diff
@@ -0,0 +1,72 @@
+from jsonschema import validate, ValidationError
+from structlog import get_logger
+
+from sqlalchemy.exc import SQLAlchemyError
+
+from application.models.models import CommunicationTemplate
+from application.utils.session_wrapper import with_db_session
+from application.utils.exceptions import InvalidTemplateException, DatabaseError
+from application.models.schema import template_schema
+
+
+logger = get_logger()
+
+UPLOAD_SUCCESSFUL = 'The upload was successful'
+INVALID_TEMPLATE = 'Template is invalid'
+PREEXISTING_TEMPLATE = 'ID already exists'
+
+
+def get_template_by_id(template_id, session):
+    try:
+        template = session.query(CommunicationTemplate).filter(CommunicationTemplate.id == template_id).first()
+    except SQLAlchemyError:
+        logger.exception("Unable to retrieve template with id: {}".format(template_id))
+        raise DatabaseError("Unable to retrieve template with id: {}".format(template_id), status_code=500)
+    return template
+
+
+def validate_template(template):
+    try:
+        validate(template, template_schema)
+    except ValidationError as exception:
+        logger.exception("Attempted to upload invalid template")
+        raise InvalidTemplateException(INVALID_TEMPLATE, status_code=400)
```

Comment:
Do we need the text `Template is invalid`? A 400 from the template upload api should indicate it's invalid? Maybe a json object containing the reasons why it's invalid would be more useful to allow the consumer to fix the issue

Reply:
I can use the ValidationError message which looks to only have validation messages in it so should be ok to expose.